docs/content/docs/data-structures/sequence.md

```diff
@@ -0,0 +1,241 @@
+be any value from 1 to the length of the sequence.
+
+```typescript
+    //   content: hi world!
+    // positions: 012345678
+
+    // remove the first 3 characters
+    sharedString.removeRange(0, 3);
+    //   content: world!
+    // positions: 012345
+
+    // remove all the characters
+    sharedString.removeRange(0, sharedString.getLength());
+    //   content:
+    // positions:
+```
+
+Annotate operations can add or remove map-like properties to or from content of the sequence. They can store any JSON
+serializable data and have similar behavior to a shared map. Annotate takes a start and end position which work the same
+way as the start and end of the remove operation. In addition to start and end annotate also takes a map-like properties
+object. Each key of the provided properties object will be set on each position of the specified range. Setting a
+property key to null will remove that property from the positions in the range.
+
+```typescript
+    //   content: hi world
+    // positions: 01234567
+
+    let props1 = sharedString.getPropertiesAtPosition(1);
+    let props5 = sharedString.getPropertiesAtPosition(5);
+    // props1 = {}
+    // props5 = {}
+
+    // set property called weight on positions 0 and 1
+    sharedString.annotateRange(0, 2, { weight: 5 });
+    props1 = sharedString.getPropertiesAtPosition(1);
+    props5 = sharedString.getPropertiesAtPosition(5);
+    // props1 = { weight: 5 }
+    // props5 = {}
+
+    // set property called decoration on all positions
+    sharedString.annotateRange(
+        0,
+        sharedString.getLength(),
+        { decoration: "underline" });
+    props1 = sharedString.getPropertiesAtPosition(1);
+    props5 = sharedString.getPropertiesAtPosition(5);
+    // props1 = { weight: 5, decoration: "underline" }
+    // props5 = { decoration: "underline" }
+
+    // remove property called weight on all positions
+    sharedString.annotateRange(
+        0,
+        sharedString.getLength(),
+        { weight: null });
+    props1 = sharedString.getPropertiesAtPosition(1);
+    props5 = sharedString.getPropertiesAtPosition(5);
+    // props1 = { decoration: "underline" }
+    // props5 = { decoration: "underline" }
+```
+
+Whenever an operation is performed on a sequence a *sequenceDelta* event will be raised. This even provides the ranges
+affected by the operation, the type of the operation, and the properties that were changes by the operation.
+
+## How Collaboration Works
+
+Like other data structures the sequences are eventually consistent which means all collaborators will end up in the same
+final state, however, the intermediate states seen by each collaborator may not be seen by other collaborators. These
+intermediate states occur when two or more collaborators modify the same position in the sequence which results in a
+conflict.
+
+The basic strategy for insert conflict resolution in the sequence is to merge *far*. This strategy depends on a
```

Comment:
I find this paragraph too complicated for describing process, and using terminology (far) that is not helping (and not used anywhere else).

I think you can simplify it by basically saying that conflicting inserts are resolved as if there was no conflict and changes were spaced out in time, with change hitting server earlier (user 1) being observed by second client but not moving his cursor (insertion point), thus making his/her (later) insert go to the left of first user's insert.

That said, does it really matter? It seems like it's implementation detail that should not leak out, unless someone builds some rather complicated layer on top of it (which is advanced topic).

Reply:
> I find right and left terminology is much more on point to describing what actually happens, and I've heard our own team using it (never heard "far" terminology).

I think one reason the near/far terminology was used is because for RTL languages, left/right isn't accurate. That was how @anthony-murphy explained it way back when. That said, I don't have a strong attachment to near/far.

I don't think any amount of words will be as helpful as some diagrams or animations of the algorithm in action if you **really** want to know what's going on under the covers. I dream of someday having those. 😄 But I think you're right - this detail shouldn't be relevant to general usage. I do love having that detail available for those who dig, but maybe that detail belongs in the contributor docs (currently the wiki)? Or we could add a "how it works" section, which has come up before. I'll mull it over.

Reply:
I do not think Sequence cares about language - resolution always happens in exactly same way no matter what is inserted (text or other types of segments like markers). And I think the meaning of "right" & "left" words is not changed in RTL languages.